src/systems/systems/src/landing_gear/mod.rs

```diff
@@ -265,55 +299,917 @@ impl LgciuGearExtension for LandingGearControlInterfaceUnit {
+        self.sensor_inputs
+            .left_gear_extended(treat_ext_pwr_as_ground)
+    }
+
+    fn left_and_right_gear_compressed(&self, treat_ext_pwr_as_ground: bool) -> bool {
+        self.sensor_inputs
+            .left_and_right_gear_compressed(treat_ext_pwr_as_ground)
+    }
+    fn left_and_right_gear_extended(&self, treat_ext_pwr_as_ground: bool) -> bool {
+        self.sensor_inputs
+            .left_and_right_gear_extended(treat_ext_pwr_as_ground)
+    }
+    fn nose_gear_compressed(&self, treat_ext_pwr_as_ground: bool) -> bool {
+        self.sensor_inputs
+            .nose_gear_compressed(treat_ext_pwr_as_ground)
+    }
+    fn nose_gear_extended(&self, treat_ext_pwr_as_ground: bool) -> bool {
+        self.sensor_inputs
+            .nose_gear_extended(treat_ext_pwr_as_ground)
+    }
+}
+impl LgciuGearExtension for LandingGearControlInterfaceUnit {
+    fn all_down_and_locked(&self) -> bool {
+        self.sensor_inputs.all_down_and_locked()
+    }
+    fn all_up_and_locked(&self) -> bool {
+        self.sensor_inputs.all_up_and_locked()
+    }
+}
+impl LgciuDoorPosition for LandingGearControlInterfaceUnit {
+    fn all_fully_opened(&self) -> bool {
+        self.sensor_inputs.all_fully_opened()
+    }
+    fn all_closed_and_locked(&self) -> bool {
+        self.sensor_inputs.all_closed_and_locked()
+    }
+}
+impl LgciuGearControl for LandingGearControlInterfaceUnit {
+    fn should_open_doors(&self) -> bool {
+        match self.gear_system_control.state() {
+            GearsSystemState::AllUpLocked => false,
+            GearsSystemState::Retracting => !self.all_up_and_locked(),
+            GearsSystemState::Extending => !self.all_down_and_locked(),
+            GearsSystemState::AllDownLocked => false,
+        }
+    }
+
+    fn should_extend_gears(&self) -> bool {
+        match self.gear_system_control.state() {
+            GearsSystemState::AllUpLocked => false,
+            GearsSystemState::Retracting => !(self.all_fully_opened() || self.all_up_and_locked()),
+            GearsSystemState::Extending => self.all_fully_opened() || self.all_down_and_locked(),
+            GearsSystemState::AllDownLocked => true,
+        }
+    }
+
+    fn control_active(&self) -> bool {
+        match self.status {
+            LgciuStatus::Ok | LgciuStatus::FailedNoChangeOver => true,
+            LgciuStatus::FailedNotPowered | LgciuStatus::FailedAutoChangeOver => false,
+        }
+    }
+}
+impl LandingGearHandle for LandingGearControlInterfaceUnit {
+    fn gear_handle_is_down(&self) -> bool {
+        self.is_gear_lever_down
+    }
+
+    fn gear_handle_baulk_locked(&self) -> bool {
+        // Unpowered lgciu will lock mechanism
+        !(self.is_powered
+            && self.left_and_right_gear_extended(false)
+            && self.nose_gear_extended(false))
+    }
+}
+
+impl LgciuInterface for LandingGearControlInterfaceUnit {}
+
+#[derive(PartialEq, Clone, Copy, Debug)]
+pub enum GearsSystemState {
```

Comment:
There is a bit of a mix between references to a "gears system state" and "gear system state". I think choosing a consistent name makes sense.

```suggestion
pub enum GearSystemState {
```